### PR TITLE
Add AuthorizationPolicy CRD definition to Pilot/Galley.

### PR DESCRIPTION
**Pull request by yangminzhu**
This PR adds the support to `AuthorizationPolicy` in Pilot/Galley. After this PR, we should be able to use the AuthorizationPolicy in `rbac.go`. 

The next step work:
1) Update `rbac.go` to select policies based on labels
2) Update Galley to convert the service name to label and synthesis the AuthorizationPolicy when necessary (POC: #11089)

### CODE REVIEW

File: pilot/pkg/model/authorization.go [hunk 1]
```diff
@@ -38,12 +40,34 @@ type RolesAndBindings struct {
...
+}
+
+func (cfg AuthorizationPolicyConfig) String() string {
+	return fmt.Sprintf("AuthorizationPolicyConfig{name: %s, policy: %s", cfg.Name, cfg.Policy)
+}
+
+// AuthorizationConfigV2 stores a list of AuthorizationPolicyConfig and ServiceRole in a given namespace.
+type AuthorizationConfigV2 struct {
+	// A list of AuthorizationPolicy.
+	AuthPolicies []*AuthorizationPolicyConfig
```
Comment (pitlv2109) [inline pilot/pkg/model/authorization.go]: AuthzPolicies (or AuthZPolicies) instead of AuthPolicies?

Comment (yangminzhu) [reply pilot/pkg/model/authorization.go]: Done.

**Comment by pitlv2109** [comment]
Thanks for the starting PR on AuthorizationPolicy Yangmin.

/lgtm
/approve

**Comment by pitlv2109** [comment]
Nice. Thanks Yangmin!

**Pull request merged**